src/mcts/node.cc

```diff
@@ -254,7 +255,7 @@ void Node::ResetStats() {
   w_ = 0.0;
   p_ = 0.0;
   max_depth_ = 0;
-  full_depth_ = 0;
+  num_leaves_ = 0;
```

Comment:
I still don't think that wasting 4 bytes to show lies is a good thing.
Maybe just show average ('real") depth of last 10000 visited nodes?

Reply:
To show.... lies? What?

And it's only +2 bytes over the previous variable

Reply:
Well, lczero also shows lies! :)
It's not real depth, but rather some arbitrary number.
What do you think about idea of showing real average of last 10000 visited nodes? Or maybe 1000.

Reply:
Maybe we should ask discord? And I wouldn't call it lies, just... only asymptotically accurate. I'm pretty sure the current PR is at least way more useful than current depth, so I would prefer to merge this now anyways and then discuss other alternatives, which I'm open to. And why not just show average of all nodes, as opposed to last n nodes? Each playout adds information.

Reply:
We can ask yes.
I don't like global average because it doesn't really show "current depth". If engine searched depth 20 for an hour, and then depth 30 for an hour, it will show 25.

Average of last 10000 is pretty straightforward to implement with a circular buffer.

What is even easier to implement, is leaky average:
```
average_depth = (1-eps) * average_depth + eps * new_node_depth;
```
With eps=1/400.0 it would be good enough.

Reply:
I think you and I have perhaps different meanings of what `depth` is supposed to be in uci output. I interpret it as a grand sum number that describes your search accomplishments to date, so 25 is perfectly reasonable to me if you have n leaves at depth 20 and n at depth 30. "Current" depth, whatever that means, is probably better off in some secondary output (`seldepth` or `info string curdepth`).

At any rate though, what I said about what's-currently-in-lc0-is-useless still applies, so can we merge this and continue this in a new issue?